src/test/regress/sql/publication.sql

```diff
@@ -0,0 +1,263 @@
+
+-- test and test-pubs will show up in worker nodes
+SELECT DISTINCT c FROM (
+  SELECT unnest(result::text[]) c
+  FROM run_command_on_workers($$
+    SELECT array_agg(c) FROM (SELECT c FROM unnest(activate_node_snapshot()) c WHERE c LIKE '%CREATE PUBLICATION%' AND c LIKE '%pubtables%' ORDER BY 1) s$$)
+  ORDER BY c) s;
+
+-- operations with a strangely named distributed table in a mixed publication
+ALTER PUBLICATION pubtables DROP TABLE "test-pubs";
+ALTER PUBLICATION pubtables ADD TABLE "test-pubs";
+
+-- create a publication with distributed and local tables
+DROP PUBLICATION pubtables;
+CREATE PUBLICATION pubtables FOR TABLE test, "test-pubs", "publication-1"."test-pubs";
+
+-- change distributed tables
+SELECT alter_distributed_table('test', shard_count := 5, cascade_to_colocated := true);
+SELECT undistribute_table('test');
+SELECT citus_add_local_table_to_metadata('test');
+SELECT create_distributed_table_concurrently('test', 'x');
+SELECT undistribute_table('"test-pubs"');
+SELECT create_reference_table('"test-pubs"');
+
+-- publications are unchanged despite various tranformations
+SELECT DISTINCT c FROM (
+  SELECT unnest(result::text[]) c
+  FROM run_command_on_workers($$
+    SELECT array_agg(c) FROM (SELECT c FROM unnest(activate_node_snapshot()) c WHERE c LIKE '%CREATE PUBLICATION%' AND c LIKE '%pubtables%' ORDER BY 1) s$$)
+  ORDER BY c) s;
+
+-- partitioned table
+CREATE TABLE testpub_partitioned (a int, b text, c text) PARTITION BY RANGE (a);
+CREATE TABLE testpub_partitioned_0 PARTITION OF testpub_partitioned FOR VALUES FROM (1) TO (10);
+ALTER TABLE testpub_partitioned_0 ADD PRIMARY KEY (a);
+ALTER TABLE testpub_partitioned_0 REPLICA IDENTITY USING INDEX testpub_partitioned_0_pkey;
+CREATE TABLE testpub_partitioned_1 PARTITION OF testpub_partitioned FOR VALUES FROM (11) TO (20);
+ALTER TABLE testpub_partitioned_1 ADD PRIMARY KEY (a);
+ALTER TABLE testpub_partitioned_1 REPLICA IDENTITY USING INDEX testpub_partitioned_1_pkey;
+CREATE PUBLICATION pubpartitioned FOR TABLE testpub_partitioned WITH (publish_via_partition_root = 'true');
+
+SELECT create_distributed_table('testpub_partitioned', 'a');
+
+SELECT DISTINCT c FROM (
+  SELECT unnest(result::text[]) c
+  FROM run_command_on_workers($$
+    SELECT array_agg(c) FROM (SELECT c FROM unnest(activate_node_snapshot()) c WHERE c LIKE '%CREATE PUBLICATION%' AND c LIKE '%pubpartitioned%' ORDER BY 1) s$$)
+  ORDER BY c) s;
+
+DROP PUBLICATION pubpartitioned;
+CREATE PUBLICATION pubpartitioned FOR TABLE testpub_partitioned WITH (publish_via_partition_root = 'true');
+
+-- add a partition
+ALTER PUBLICATION pubpartitioned ADD TABLE testpub_partitioned_1;
+
+SELECT DISTINCT c FROM (
+  SELECT unnest(result::text[]) c
+  FROM run_command_on_workers($$
+    SELECT array_agg(c) FROM (SELECT c FROM unnest(activate_node_snapshot()) c WHERE c LIKE '%CREATE PUBLIATION%' AND c LIKE '%pubpartitioned%' ORDER BY 1) s$$)
+  ORDER BY c) s;
+
+DROP PUBLICATION pubempty;
```

Comment:
maybe before dropping these call `start_metadata_sync_to_all_nodes()` here to ensure we can re-sync these publications without a problem.